src/Extractor/LinkCollectionExtractor.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * @license   http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ * @copyright Copyright (c) 2014 Zend Technologies USA Inc. (http://www.zend.com)
+ */
+
+namespace ZF\Hal\Extractor;
+
+use ZF\ApiProblem\Exception\DomainException;
+use ZF\Hal\Link\Link;
+use ZF\Hal\Link\LinkCollection;
+
+class LinkCollectionExtractor implements LinkCollectionExtractorInterface
+{
+    /**
+     * @var LinkExtractorInterface
+     */
+    protected $linkExtractor;
+
+    /**
+     * @param LinkExtractorInterface $linkExtractor
+     */
+    public function __construct(LinkExtractorInterface $linkExtractor)
+    {
+        $this->setLinkExtractor($linkExtractor);
+    }
+
+    /**
+     * @return LinkExtractorInterface
+     */
+    public function getLinkExtractor()
+    {
+        return $this->linkExtractor;
+    }
+
+    /**
+     * @param LinkExtractorInterface $linkExtractor
+     */
+    public function setLinkExtractor(LinkExtractorInterface $linkExtractor)
+    {
+        $this->linkExtractor = $linkExtractor;
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
There was an interesting exchange over Twitter about this last week. `{@inheritdoc}` will only inherit the description; you need to use `@inheritdoc` to inherit also the parameter/return value definitions. (I can fix this on merge.)